tests/main/netplan-apply/task.yaml

```diff
@@ -0,0 +1,74 @@
+summary: Ensure that netplan apply works with network-setup-control
+
+details: |
+    Netplan apply is used to apply network configuration to the system
+
+# run on all classic ubuntu LTS and current dev systems 16.04+
+systems: 
+    - ubuntu-16*
+    - ubuntu-18*
+    - ubuntu-19*
+    - ubuntu-20*
+
+prepare: |
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    snap install test-snapd-netplan-apply --edge
+
+    # backup the dbus service file and policy config if they exist before 
+    # executing
+    for f in system-services/io.netplan.Netplan.service system.d/io.netplan.Netplan.conf; do
+        if [ -f /usr/share/dbus-1/$f ]; then
+            mv /usr/share/dbus-1/$f /usr/share/dbus-1/$f.backup
+        fi      
+    done
+
+restore: |
+    # restore the dbus service file and policy config file if the backup 
+    # exists
+    for f in system-services/io.netplan.Netplan.service system.d/io.netplan.Netplan.conf; do
+        if [ -f /usr/share/dbus-1/$f.backup ]; then
+            mv /usr/share/dbus-1/$f.backup /usr/share/dbus-1/$f
+        fi      
+    done
+
+execute: |
+    # install the dbus configuration file for our fake netplan system dbus service
+    mkdir -p /usr/share/dbus-1/system.d 
+    mkdir -p /usr/share/dbus-1/system-services
+    cp io.netplan.Netplan.conf /usr/share/dbus-1/system.d/io.netplan.Netplan.conf
+
+    # install the fake netplan system dbus service
+    echo "Make the netplan service activatable"
+    cat << EOF > /usr/share/dbus-1/system-services/io.netplan.Netplan.service
+    [D-BUS Service]
+    Name=io.netplan.Netplan
+    Exec=$(pwd)/fake-netplan-apply-service.py $(pwd)/dbus-netplan-apply.log
```

Comment:
We may have issues with this on SELinux systems, due to tag based permissions. At the same time I don't have a workaround I can present.